lib/rules/padding-between-test-functions.js

```diff
@@ -0,0 +1,142 @@
+            ? 'Test functions must not be padded by blank lines.'
+            : 'Test functions must be padded by blank lines.',
+        sourceCode = context.getSourceCode();
+
+        return {
+            ExpressionStatement: function (node) {
+                var firstToken = sourceCode.getFirstToken(node),
+                    tokenBeforeFirst = sourceCode.getTokenBefore(firstToken, { includeComments: true }),
+                    lastToken = sourceCode.getLastToken(node),
+                    tokenAfterLast = sourceCode.getTokenAfter(lastToken, { includeComments: true });
+
+                function shouldReturn() {
+                    return (
+                        !sourceCode.getTokenBefore(node) ||
+                        !sourceCode.getTokenAfter(node) ||
+                        !isTestFunction(node)
+                    );
+                }
+
+                function always() {
+                    var blockHasTopPadding = isPaddingBetweenTokens(tokenBeforeFirst, firstToken),
+                        blockHasBottomPadding = isPaddingBetweenTokens(lastToken, tokenAfterLast);
+                    if (!blockHasTopPadding) {
+                        context.report({
+                            node: node,
+                            message: message,
+                            fix: function (fixer) {
+                                return fixer.insertTextBefore(firstToken, '\n');
+                            }
+                        });
+                    }
+                    if (!blockHasBottomPadding) {
+                        context.report({
+                            node: node,
+                            message: message,
+                            fix: function (fixer) {
+                                return fixer.insertTextAfter(lastToken, '\n');
+                            }
+                        });
+                    }
+                }
+
+                function never() {
+                    var blockHasTopPadding = isPaddingBetweenTokens(tokenBeforeFirst, firstToken),
+                        blockHasBottomPadding = isPaddingBetweenTokens(lastToken, tokenAfterLast);
+                    if (blockHasTopPadding) {
+                        context.report({
+                            node: node,
+                            message: message,
+                            fix: function (fixer) {
+                                return fixer.replaceTextRange([ tokenBeforeFirst.range[1], firstToken.range[0] ], '\n');
+                            }
+                        });
+                    }
+                    if (blockHasBottomPadding) {
+                        context.report({
+                            node: node,
+                            message: message,
+                            fix: function (fixer) {
+                                return fixer.replaceTextRange([ lastToken.range[1], tokenAfterLast.range[0] ], '\n');
+                            }
```

Comment:
Looks like there is a little bit of code duplication in those fixer functions. What do you think about extracting this to a function that creates line removal fixer functions for the given tokens?

Example:

```js
function removeEmptyLinesBetweenTokens(first, second) {
    return function (fixer) {
        return fixer.replaceTextRange([ first.range[1], second.range[0] ], '\n');
    };
}
```

Reply:
Good call, I'll do that.